workflow/train_test_split.py

```diff
@@ -0,0 +1,167 @@
+from collections.abc import Iterable
+import numpy as np
+import pandas as pd
+import pytest
+
+
+def train_test_split(
+    df,
+    unique_key,
+    test_size=0.2,
+    seed=0,
+    stratify=None
+):
+    if stratify is None:
+        test_df = get_test_split(df, unique_key, test_size, seed)
+    else:
+        test_df = stratified_split(df, unique_key, stratify, test_size, seed)
+
+    train_df = df[~df[unique_key].isin(test_df[unique_key])]
+
+    return train_df, test_df
+
+
+def get_test_split(df, unique_key, test_size, seed=0, resolution=100000):
+    split_unique_key = df[unique_key].apply(
+        lambda x: ((hash(hash(x) + seed) % resolution))
+    )
+    return df[split_unique_key < resolution * test_size]
+
+
+def stratified_split(df, unique_key, stratify, test_size, seed):
+
+    test_dfs = []
+    for label in get_labels(df, stratify):
+        if type(stratify) is str:
+            df_part = df[df[stratify] == label]
+        else:
+            df_part = df[df[label].notnull()]
+        test_dfs.append(get_test_split(df_part, unique_key, test_size, seed))
+        df = df[~df[unique_key].isin(df_part[unique_key])]
+    assert len(df) == 0, f'Unable to stratify on {stratify}'
```

Comment:
tried to reason about this 3 times :P
what is the minimum number of labels that can end up in test split without it asserting?